### DISCUSSION TITLE
V2.0 Upgrades - General Discussion

### DISCUSSION THREAD

**Discussion by phil-6** [excid3/noticed#387] [General]
Hey All

Just creating a discussion around people V2.0 Upgrades. I'm most of the way through ours now and it's been a lot more work than I expected.

The PR is at 116 files changed with 1900 additions and 2100 removals. :scream: 

But, overall I love the direction and we're already captializing on some of the new capabilities. 

Edit
Final PR is 130 files, 1800 additions and 1800 removals :see_no_evil:

**Comment by excid3** [comment]
What are the biggest changes? I was thinking of a few things we could do to make the migration easier. For example, re-adding `param` with a deprecation warning.

**Reply by phil-6** [reply]
Rewriting the notification models to split out custom logic. We had 11 notification types (and are about to add about 6 more)

But, once I'd figured it out for one, all the others were pretty straightforward, it did feel like I had to re-write all the notifiers from scratch.

There were a LOT of little changes to fix and across the app. 

I'd be happy to give you a walkthrough of our setup if that might be helpful.

**Reply by excid3** [reply]
Yeah, I knew those were going to be a bit painful since we split the models out. Anything stand out as particularly tricky?

And yes, would love to that. Shoot me an email?

**Reply by phil-6** [reply]
You're email is hard to find! Fired you a message on twitter :)

**Reply by excid3** [reply]
I dunno, it's in almost every screencast. 😜
